astropy/timeseries/periodograms/lombscargle/implementations/fast_impl.py

```diff
@@ -14,11 +14,12 @@ def lombscargle_fast(
     fit_mean=True,
     normalization="standard",
     use_fft=True,
+    prefer_lra=True,
     trig_sum_kwds=None,
```

Comment:
any new argument to a public API function needs to be added at the *end* of the signature for backward compatibility.  Even if this module is deeply nested and therefore officially private, it is not clearly marked as such so, no reason to break backward compatibility when the cost of preserving it is so low. It is also good practice to avoid adding positional-allowed arguments beyond 3 or 4 args.
```suggestion
    trig_sum_kwds=None,
    *,
    prefer_lra=True,
```

Reply:
Additionally, I'd recommend we avoid creating another [boolean trap](https://adamj.eu/tech/2021/07/10/python-type-hints-how-to-avoid-the-boolean-trap/). Can this be expressed as `method='low-rank-approximation'` VS `method='press-rybicky'` (which would be the default) ?